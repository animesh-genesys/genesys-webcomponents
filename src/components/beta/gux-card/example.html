<h1>gux-card-beta</h1>
<h2>Outline Card</h2>
<gux-card-beta>
  <h2>Outcome Details</h2>
  <gux-form-field-text-like label-position="above">
    <input
      slot="input"
      type="text"
      name="a-2"
      placeholder="Please enter a name"
      required
    />
    <label slot="label">Name</label>
  </gux-form-field-text-like>
  <gux-form-field-text-like label-position="above">
    <input
      slot="input"
      type="text"
      name="a-2"
      placeholder="Please enter a description"
    />
    <label slot="label">Description</label>
  </gux-form-field-text-like>
  <gux-card-beta accent="filled">
    <div class="card-content">
      <label>Status</label>
      <gux-toggle
        checked-label="The outcome is active."
        unchecked-label="The outcome is not active."
      ></gux-toggle>
    </div>
  </gux-card-beta>
</gux-card-beta>
<h2>Outline Card (Explicity Set)</h2>
<gux-card-beta accent="outline">
  <h2>Define your outcome conditions</h2>
  <gux-card-beta accent="filled">
    <p>What activity type should be used to achieve this outcome?</p>
    <gux-form-field-select label-position="beside">
      <select slot="input" name="lp-1">
        <option value="option1">Session 1</option>
        <option value="option2">Session 2</option>
        <option value="option3">Session 3</option>
      </select>
      <label slot="label">Session Type</label>
    </gux-form-field-select>
  </gux-card-beta>
  <gux-button class="button-styling">Add Event</gux-button>
</gux-card-beta>
<h2>Filled Card</h2>
<gux-card-beta accent="filled">
  <p>What session type is required for this action to be performed?</p>
  <gux-form-field-select label-position="beside">
    <select slot="input" name="lp-1">
      <option value="option1">Session 1</option>
      <option value="option2">Session 2</option>
      <option value="option3">Session 3</option>
    </select>
    <label slot="label">Session Type</label>
  </gux-form-field-select>
</gux-card-beta>
<h2>Raised Card</h2>
<gux-card-beta accent="raised">
  <h2>Define your Visitor Characteristics</h2>
  <gux-card-beta accent="filled">
    <p>All characteristics must be met for outcome to be achieved.</p>
    <gux-button>Add Characteristic</gux-button>
  </gux-card-beta>
</gux-card-beta>

<style>
  gux-form-field-text-like {
    width: 700px;
  }

  .card-content {
    display: flex;
    flex-direction: row;
    gap: 20px;
    align-items: center;
    justify-content: flex-start;
  }

  .button-styling {
    margin-top: 8px;
  }
</style>
