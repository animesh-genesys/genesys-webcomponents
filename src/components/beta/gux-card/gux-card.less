@import (reference) '../../../style/color.less';
@import (reference) '../../../style/mixins.less';
@import (reference) '../../../style/spacing.less';
@import (reference) '../../../style/typography.less';
@import (reference) '../../../style/shadows.less';

:host {
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  width: fit-content;
  padding: @spacing-medium;
  background-color: @gux-grey-100;
  border: 1px solid @gux-grey-60;
  border-radius: @spacing-2xs;
}

:host([accent='outline']) {
  background-color: @gux-grey-100;
}

:host([accent='filled']) {
  background-color: @gux-grey-90;
}

:host([accent='raised']) {
  .gux-shadow-10();
}
