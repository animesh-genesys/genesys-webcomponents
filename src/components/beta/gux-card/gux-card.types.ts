export type GuxCardAccent = 'outline' | 'filled' | 'raised';
