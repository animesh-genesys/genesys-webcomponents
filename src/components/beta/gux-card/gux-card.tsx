import { Component, h, JSX, Prop, Element } from '@stencil/core';
import { trackComponent } from '../../../usage-tracking';

import { GuxCardAccent } from './gux-card.types';

/**
 * @slot - Content of card.
 */

@Component({
  styleUrl: 'gux-card.less',
  tag: 'gux-card-beta',
  shadow: true
})
export class GuxCard {
  /*
   * Reference to the host element.
   */
  @Element()
  root: HTMLElement;

  /**
   * Card Accent.
   */
  @Prop()
  accent: GuxCardAccent = 'outline';

  componentWillLoad(): void {
    trackComponent(this.root);
  }

  render(): JSX.Element {
    return (<slot />) as JSX.Element;
  }
}
