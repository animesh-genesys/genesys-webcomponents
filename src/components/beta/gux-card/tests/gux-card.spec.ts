import { newSpecPage } from '@stencil/core/testing';
import { GuxCard } from '../gux-card';

const components = [GuxCard];
const language = 'en';

describe('#render different card variations eg(outline,outline(explicit),filled,raised)', () => {
  [
    '<gux-card-beta><span>This is a content slot</span></gux-card-beta>',
    '<gux-card-beta accent="outline"><span>This is a content slot</span></gux-card-beta>',
    '<gux-card-beta accent="raised"><span>This is a content slot</span></gux-card-beta>',
    '<gux-card-beta accent="filled"><span>This is a content slot</span></gux-card-beta>'
  ].forEach((html, index) => {
    it(`should render component as expected (${index + 1})`, async () => {
      const page = await newSpecPage({ components, html, language });

      expect(page.root).toMatchSnapshot();
    });
  });
});
