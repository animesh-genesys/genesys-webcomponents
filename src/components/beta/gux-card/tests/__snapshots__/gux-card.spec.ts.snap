// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render different card variations eg(outline,outline(explicit),filled,raised) should render component as expected (1) 1`] = `
<gux-card-beta>
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <span>
    This is a content slot
  </span>
</gux-card-beta>
`;

exports[`#render different card variations eg(outline,outline(explicit),filled,raised) should render component as expected (2) 1`] = `
<gux-card-beta accent="outline">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <span>
    This is a content slot
  </span>
</gux-card-beta>
`;

exports[`#render different card variations eg(outline,outline(explicit),filled,raised) should render component as expected (3) 1`] = `
<gux-card-beta accent="raised">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <span>
    This is a content slot
  </span>
</gux-card-beta>
`;

exports[`#render different card variations eg(outline,outline(explicit),filled,raised) should render component as expected (4) 1`] = `
<gux-card-beta accent="filled">
  <mock:shadow-root>
    <slot></slot>
  </mock:shadow-root>
  <span>
    This is a content slot
  </span>
</gux-card-beta>
`;
